Buildings/Applications/DHC/EnergyTransferStations/CoolingIndirect.mo

```diff
@@ -0,0 +1,287 @@
+    annotation (Placement(
+        transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={50,-130}), iconTransformation(
+        extent={{-10,-10},{10,10}},
+        origin={30,-110},
+        rotation=-90)));
+
+  Modelica.Blocks.Continuous.Integrator int(k=1) "Integration"
+    annotation (Placement(transformation(extent={{0,-120},{20,-100}})));
+
+  Buildings.Fluid.Sensors.MassFlowRate senMasFlo(redeclare package Medium = Medium)
+    annotation (Placement(transformation(extent={{-60,50},{-40,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort TBuiRet(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m2_flow_nominal)
+    "Building-side (secondary) return temperature"
+    annotation (Placement(transformation(extent={{-70,-70},{-90,-50}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayQuickOpening val(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m1_flow_nominal,
+    dpValve_nominal=dpValve_nominal,
+    riseTime(displayUnit="s") = 60,
+    y_start=0) "District-side (primary) control valve"
+    annotation (Placement(transformation(extent={{-30,70},{-10,50}})));
+
+equation
+
+  connect(hex.port_a2, port_a2) annotation (Line(points={{40,-6},{60,-6},{60,
+          -60},{100,-60}},
+                      color={0,127,255}));
+  connect(hex.port_b1, senTDisRet.port_a) annotation (Line(points={{40,6},{60,6},
+          {60,60},{70,60}}, color={0,127,255}));
+  connect(val.port_b, hex.port_a1) annotation (Line(points={{-10,60},{0,60},{0,
+          6},{20,6}},  color={0,127,255}));
+  connect(senMasFlo.port_b, val.port_a)
+    annotation (Line(points={{-40,60},{-30,60}}, color={0,127,255}));
+  connect(port_a1, senTDisSup.port_a)
+    annotation (Line(points={{-100,60},{-90,60}}, color={0,127,255}));
+  connect(senTDisSup.port_b, senMasFlo.port_a)
+    annotation (Line(points={{-70,60},{-60,60}}, color={0,127,255}));
+  connect(port_b2, TBuiRet.port_b)
+    annotation (Line(points={{-100,-60},{-90,-60}}, color={0,127,255}));
+  connect(senTDisRet.port_b, port_b1)
+    annotation (Line(points={{90,60},{100,60}}, color={0,127,255}));
+  connect(powCal.y, Q_flow)
+    annotation (Line(points={{-16,-90},{90,-90},{90,-130}}, color={0,0,127}));
+  connect(powCal.y, int.u) annotation (Line(points={{-16,-90},{-10,-90},{-10,-110},
+          {-2,-110}}, color={0,0,127}));
+  connect(int.y, Q)
+    annotation (Line(points={{21,-110},{50,-110},{50,-130}}, color={0,0,127}));
+  connect(TSet, con.u_s)
+    annotation (Line(points={{-120,0},{-92,0}}, color={0,0,127}));
+  connect(con.u_m, TBuiRet.T)
+    annotation (Line(points={{-80,-12},{-80,-49}}, color={0,0,127}));
+  connect(con.y, val.y)
+    annotation (Line(points={{-69,0},{-20,0},{-20,48}}, color={0,0,127}));
+  connect(TBuiRet.port_a, hex.port_b2) annotation (Line(points={{-70,-60},{0,-60},
+          {0,-6},{20,-6}}, color={0,127,255}));
+  annotation (Icon(coordinateSystem(preserveAspectRatio=false), graphics={
```

Comment:
the class should have a default component name. This can be done by adding
`defaultComponentName=xxx`.

Reply:
Done